### ISSUE TITLE
terminal.integrated.cwd setting does not work, and does not default to workspace folder

### ISSUE THREAD

**Issue by JPlumridge** [microsoft/vscode#75759]
- VSCode Version: **1.34.0** (system setup)
- OS Version: Microsoft Windows 10 Pro (**10.0.17134 Build 17134**)

### Steps to Reproduce:

1. Load VS code in a specific directory
2. Ctrl + Shift + P > select "Terminal: Focus Terminal"
3. Terminal opens in the parent folder of the current folder, NOT the current folder

i.e folder open is "C:\Repositories\MyProject". Terminal opens in "C:\Repositories"

Does this issue occur when all extensions are disabled?: **Yes**

### Things I have tried:

Setting "terminal.integrated.cwd " to "${workspaceFolder}"
Result: terminal fails to load completely (new issue for this?)

Deleting the terminal instance with 🗑 icon
Result: no change

Restarting VSCode:
Result: no change

Setting "terminal.integrated.cwd " to "C:\Repositories\MyProject"
Result: terminal fails to load completely

Creating a "workspace" instead of just opening a folder
Result: no change

Changing default shell from PowerShell to Git Bash
Result: no change

All of the above tried in various combinations.

**Comment by JPlumridge** [comment]
I've just tested this on 1.35.1 (I thought I'd updated before raising this, my fault).

Small change:

Setting "terminal.integrated.cwd " to "${workspaceFolder}"
Result: terminal pane loads but no terminal actually loads, but there's no error message saying failed to open terminal as on 1.34.0

**Comment by Tyriar** [comment]
I can't reproduce this, do you see it in other folders as well? Not setting `.cwd` should open it in the currently opened folder.

Are you in a multi-root workspace?

**Comment by msftmalau** [comment]
I noticed that this `"terminal.integrated.cwd"` does not work with Powershell Core on which I have installed on `"C:\\Program Files\\PowerShell\\6\\pwsh.exe"`.

This settings works fine for Windows Powershell ("C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"`), as well as Windows Subsystem Linux or Git Bash.

Maybe @JPlumridge has Core installed?

Hopefully, this could lead to a fix.

**Comment by Tyriar** [comment]
Works fine for me with pwsh? Tested with:

```
"terminal.integrated.cwd": "${fileDirname}",
"terminal.integrated.shell.windows": "C:\\Program Files\\PowerShell\\6\\pwsh.exe"
```

**Comment by msftmalau** [comment]
It might be a different issue then. My VSCode (v1.37.0) does not recognize `"${fileDirname}"` for both versions of Powershell.

```
The terminal shell CWD "${fileDirname}" does not exist
```

I can see that these predefined variables should exist: https://vscode.trafficmanager.net/docs/editor/variables-reference

But where are they defined?

**Comment by Tyriar** [comment]
@msftmalau open dev tools via the help menu and check the console for an error, you might not have a file opened?

**Comment by msftmalau** [comment]
Apologies. Turns out I had a Powershell Profile set up to point to `$HOME`. I previously set this because "Run as Administrator" was pointing me to `C:\WINDOWS\system32` instead of `$HOME` or where I want to be.

`"terminal.integrated.cwd": "${fileDirname}"` is working as expected.
Not setting `"terminal.integrated.cwd"` also points to the current working directory.

This might be separate issue now. How should I set up my Powershell Profile so that it doesn't affect VSCode integrated Powershell?
